packages/flutter/lib/src/material/drawer.dart

```diff
@@ -261,13 +316,22 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
     _controller.fling(velocity: -1.0);
   }
 
+  bool isOpen() {
+    if (_controller.value >= 1.0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
   final ColorTween _color = new ColorTween(begin: Colors.transparent, end: Colors.black54);
   final GlobalKey _gestureDetectorKey = new GlobalKey();
 
   Widget _buildDrawer(BuildContext context) {
     if (_controller.status == AnimationStatus.dismissed) {
       return new Align(
-        alignment: AlignmentDirectional.centerStart,
+        alignment: (widget.type == DrawerType.START) ? 
+          AlignmentDirectional.centerStart : AlignmentDirectional.centerEnd,
```

Comment:
I'd break these out into private getters and use a switch statement instead of `operator==` on the enum.

Reply:
Fixed